lib/ex_aws/rds/impl.ex

```diff
@@ -0,0 +1,147 @@
+defmodule ExAws.RDS.Impl do
+  import ExAws.Utils, only: [camelize_keys: 1, upcase: 1]
+
+  @version "2014-10-31"
+
+  def add_source_id_to_subscription(client, source_id, subscription) do
+    query_params = %{
+      "Action"           => "AddSourceIdentifierToSubscription",
+      "SourceIdentifier" => source_id,
+      "SubscriptionName" => subscription,
+      "Version"          => @version
+    }
+
+    request(client, :post, "/", params: query_params)
+  end
+
+  def add_tags_to_resource(client, resource, tags, n) do
+    query_params = %{
+      "Action" => "AddTagsToResource",
+      "ResourceName" => resource,
+      "Version" => @version
+    }
+
+    query_params =
+    for {k, v} <- tags do 
+      query_params
+      |> Map.put_new("Tags.member.#{Integer.to_string(n)}.Key", k)
+      |> Map.put_new("Tags.member.#{Integer.to_string(n)}.Value", v)
```

Comment:
Hey! so this is almost right. The way this works is that suppose I have tags `[foo: "bar", baz: "buz"]` what I want to create is

```
%{
  "Tags.Member.0.Key" => "foo",
  "Tags.Member.0.Value" => "bar",
  "Tags.Member.1.Key" => "baz",
  "Tags.Member.1.Value" => "buz",
}
```

So you don't need them to pass in an `n` value, instead just use Stream.with_index in your for comprehension.

As usual, just use `Map.put` here instead of `put_new`, there's no chance it's already in the map.